src/server/pkg/deploy/assets/assets.go

```diff
@@ -529,6 +531,60 @@ func GetBackendSecretVolumeAndMount(backend string) (v1.Volume, v1.VolumeMount)
 		}
 }
 
+// GetPachctlSecretVolumeAndMount returns a Volume and
+// VolumeMount object configured for the pachctl secret (currently used in spout pipelines).
+func GetPachctlSecretVolumeAndMount(secret string) (v1.Volume, v1.VolumeMount) {
+	return v1.Volume{
+			Name: client.PachctlSecretName,
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: secret,
+				},
+			},
+		}, v1.VolumeMount{
+			Name:      client.PachctlSecretName,
+			MountPath: "/pachctl",
+		}
+}
+
+// CreatePachctlSecret returns a Secret configured for the pachctl secret (currently used in spout pipelines).
+func CreatePachctlSecret(token, secret string) (*v1.Secret, error) {
```

Comment:
I want to defer on someone who knows our codebase structure better - I kind of feel like this should like in the worker_rc file after all?